getting_started/step_by_step/resources.rst

```diff
@@ -158,12 +158,173 @@ Freeing resources
+
+- They can define constants, so constants from other data fields or objects are not needed.
+
+- They can define methods, including setter/getter methods for properties. This allows for abstraction and encapsulation of the underlying data. If the Resource script's structure needs to change, the game using the Resource need not also change.
+
+- They can define signals, so Resources can trigger responses to changes in the data they manage.
+
+- They have defined properties, so users know 100% that their data will exist.
+
+- Resource auto-serialization and deserialization is a built-in Godot Engine feature. Users do not need to implement custom logic to import/export a resource file's data.
+
+- Resources can even serialize sub-Resources recursively, meaning users can design even more sophisticated data structures.
+
+- Users can save Resources as version-control-friendly text files (\*.tres). Upon exporting a game, Godot serializes resource files as binary files (\*.res) for increased compression.
+
+- Godot Engine's Inspector renders and edits Resource files out-of-the-box. As such, users do not need to implement custom logic to visualize or edit their data. To do so, double-click the resource file in the FileSystem dock or click the folder icon in the Inspector and open the file in the dialog.
+
+- They can extend **other** resource types besides just the base Resource.
+
+Godot makes it easy to create custom Resources in the Inspector.
+
+1. Create a plain Resource object in the Inspector. This can even be a type that derives Resource, so long as your script is extending that type.
+2. Set the ``script`` property in the Inspector to be your script.
+
+The Inspector will now display your Resource script's custom properties. If one edits
+those values and saves the resource, the Inspector serializes the custom properties
+too! To save a resource from the Inspector, click the Inspector's tools menu (top right),
+and select "Save" or "Save As...".
+
+If the script's language supports :ref:`script classes <register-scripts-as-classes>`, 
+then it streamlines the process. Defining a name for your script alone will add it to
+the Inspector's creation dialog. This will auto-add your script to the Resource
+object you create.
+
+Let's see some examples.
+
+.. tab::
+  ..code-tab:: gdscript GDScript
+    # bot_stats.gd
+    extends Resource
+    export(int) var health
+    export(Resource) var sub_resource
+    export(Array, String) var str_array
+
+    func _init(p_health = 0, p_sub_resource = null, p_str_array = []):
+        health = p_health
+        sub_resource = p_sub_resource
+        str_array = p_str_array
+
+    # bot.gd
+    extends KinematicBody
+    export(Resource) var stats
+
+    const BotStats = preload("res://bot_stats.gd")
+
+    func _init():
+        if not stats:
+            stats = BotStats.new(10)
+    
+    func _ready():
+        print(stats.health) # prints '10'
+  ..code-tab:: csharp
+        using Godot;
```

Comment:
I think you forgot to add a C# version of `bot_stats.gd`.

Reply:
Oh yeah, guess I could add that. Like I said though, I doubt any of the C# code is really accurate since I haven't used Godot C# at all.

Reply:
Since you added a C# version, mind resolving this conversation? I can't seem to do so myself.